evap/results/tests/test_views.py

```diff
@@ -514,37 +514,37 @@ def test_textanswer_visibility_for_student(self):
         # the external user does not participate in or contribute to the evaluation and therefore can't see the results
-        self.app.get("/results/semester/1/evaluation/1", user='student_external', status=403)
+        self.app.get("/results/semester/1/evaluation/1", user="student_external@example.com", status=403)
 
     def test_textanswer_visibility_info_is_shown(self):
-        page = self.app.get("/results/semester/1/evaluation/1", user='contributor')
+        page = self.app.get("/results/semester/1/evaluation/1", user="contributor@institution.example.com")
         self.assertRegex(page.body.decode(), r"can be seen by:<br />\s*contributor user")
 
     def test_textanswer_visibility_info_for_proxy_user(self):
-        page = self.app.get("/results/semester/1/evaluation/1", user='responsible')
+        page = self.app.get("/results/semester/1/evaluation/1", user="responsible@institution.example.com")
         self.assertIn("responsible_contributor user (1 person)", page)
 
 
 class TestResultsOtherContributorsListOnExportView(WebTest):
     @classmethod
     def setUpTestData(cls):
         cls.semester = baker.make(Semester, id=2)
-        responsible = baker.make(UserProfile, username='responsible')
+        responsible = baker.make(UserProfile, email='responsible@institution.example.com')
         cls.evaluation = baker.make(Evaluation, id=21, state='published', course=baker.make(Course, semester=cls.semester, responsibles=[responsible]))
 
         questionnaire = baker.make(Questionnaire)
         baker.make(Question, questionnaire=questionnaire, type=Question.LIKERT)
         cls.evaluation.general_contribution.questionnaires.set([questionnaire])
 
         baker.make(Contribution, evaluation=cls.evaluation, contributor=responsible, questionnaires=[questionnaire], can_edit=True, textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS)
-        cls.other_contributor_1 = baker.make(UserProfile, username='other contributor 1')
+        cls.other_contributor_1 = baker.make(UserProfile, email='other_contributor_1@institution.example.com')
         baker.make(Contribution, evaluation=cls.evaluation, contributor=cls.other_contributor_1, questionnaires=[questionnaire], textanswer_visibility=Contribution.TextAnswerVisibility.OWN_TEXTANSWERS)
-        cls.other_contributor_2 = baker.make(UserProfile, username='other contributor 2')
+        cls.other_contributor_2 = baker.make(UserProfile, email='other_contributor_2@institution.example.com')
         baker.make(Contribution, evaluation=cls.evaluation, contributor=cls.other_contributor_2, questionnaires=[questionnaire], textanswer_visibility=Contribution.TextAnswerVisibility.OWN_TEXTANSWERS)
 
     def test_contributor_list(self):
         url = '/results/semester/{}/evaluation/{}?view=export'.format(self.semester.id, self.evaluation.id)
-        page = self.app.get(url, user='responsible')
+        page = self.app.get(url, user="responsible@institution.example.com")
```

Comment:
Not in the scope of this PR

Here, it would be much easier if we changed line 532 to
```python
cls.responsible = baker.make(UserProfile, email='responsible@institution.example.com')
```
so we can use
```python
page = self.app.get(url, user=self.responsible)
```
here.

It seems like we have this case a few times: The UserProfile instance is available or can easily be made available, yet we still use the username, which is rather unstable in my opinion. Should I add an issue about this?

Reply:
see #1475